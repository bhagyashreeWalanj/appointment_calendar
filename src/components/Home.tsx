import { useState, useEffect } from 'react'
import { FaClock } from 'react-icons/fa'
import { ITimeSlot } from '../models/ITimeSlots'
import TimeSlots from './TimeSlots'
import Calendar from './Calendar'
import styles from '../styles/Home.module.scss'

interface IHome {
  timeSlots: ITimeSlot
  fetchAllSlots(): void
  scheduleCall(): void
  mentor: string
}

const Home = ({ timeSlots, fetchAllSlots, mentor, scheduleCall }: IHome) => {
  const todayDateTime = new Date()
  const [date, setDate] = useState(todayDateTime)

  const handleSelectedDay = (selectedDay: any) => {
    setDate(selectedDay)
  }

  useEffect(() => {
    fetchAllSlots()
  }, [fetchAllSlots])

  return (
    <>
      <div className={styles.event}>
        {/* left container */}
        <div className={styles.event__mentor}>
          <div className={styles.event__mentor_name}>
            <label id={'mentor_name'}>{mentor}</label>
          </div>
          <div className={styles.event__mentor_title}>60 Minutes Meeting</div>
          <div className={styles.event__mentor_hours}>
            <span className={styles.event__mentor_hours_clock_icon}>
              <FaClock /> 1 Hr
            </span>
          </div>
        </div>
        {/* Calendar container */}
        <div className={styles.event__calendar}>
          <h2>Select Date and Time</h2>
          <Calendar selectedDayFn={(day) => handleSelectedDay(day)} />
        </div>
        {/* TimeSlot Container */}
        {Object.keys(timeSlots).length !== 0 ? (
          <div className={styles.event_timeslot}>
            <TimeSlots
              selectedDate={date}
              slots={timeSlots}
              mentor={mentor}
              scheduleCallFn={scheduleCall}
            />
          </div>
        ) : (
          ''
        )}
      </div>
    </>
  )
}

export default Home
