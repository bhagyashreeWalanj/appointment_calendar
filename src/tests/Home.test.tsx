import React from 'react'
import { shallow } from 'enzyme'
import Home from '../components/Home'

describe('Testing Home Component', () => {
  const fetchAllSlots = jest.fn()
  const scheduleCall = jest.fn()
  const timeSlots = {
    mentor: {
      name: 'Max Mustermann',
      time_zone: '-03:00',
    },
    calendar: [
      {
        date_time: '2022-07-24 00:37:02 +0200',
      },
      {
        date_time: '2022-07-24 03:31:50 +0200',
      },
      {
        date_time: '2022-07-24 12:13:56 +0200',
      },
      {
        date_time: '2022-07-25 20:49:10 +0200',
      },
    ],
  }

  const wrapper = shallow(
    <Home
      timeSlots={timeSlots}
      fetchAllSlots={fetchAllSlots}
      mentor="Bhagyashree WALANJ"
      scheduleCall={scheduleCall}
    />,
  )

  const mentorContainer = wrapper.find('#mentor_name')

  it('should display to Home Component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should display mentor name ', () => {
    expect(mentorContainer.text().includes('Bhagyashree')).toBe(true)
  })
})
