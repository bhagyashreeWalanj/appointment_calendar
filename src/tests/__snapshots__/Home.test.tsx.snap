// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home Component should display to Home Component 1`] = `
<Fragment>
  <div
    className="event"
  >
    <ToastContainer
      autoClose={5000}
      closeButton={[Function]}
      closeOnClick={true}
      draggable={true}
      draggableDirection="x"
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      role="alert"
      rtl={false}
      theme="light"
      transition={[Function]}
    />
    <div
      className="event__mentor"
    >
      <div
        className="event__mentor_name"
      >
        <label
          id="mentor_name"
        >
          Bhagyashree WALANJ
        </label>
      </div>
      <div
        className="event__mentor_title"
      >
        60 Minutes Meeting
      </div>
      <div
        className="event__mentor_hours"
      >
        <span
          className="event__mentor_hours_clock_icon"
        >
          <FaClock />
           1 Hr
        </span>
      </div>
    </div>
    <div
      className="event__calendar"
    >
      <h2>
        Select Date and Time
      </h2>
      <Calendar
        selectedDayFn={[Function]}
      />
    </div>
    <div
      className="event_timeslot"
    >
      <TimeSlots
        mentor="Bhagyashree WALANJ"
        scheduleCallFn={[MockFunction]}
        selectedDate={2022-07-17T20:54:25.347Z}
        slots={
          Object {
            "calendar": Array [
              Object {
                "date_time": "2022-07-24 00:37:02 +0200",
              },
              Object {
                "date_time": "2022-07-24 03:31:50 +0200",
              },
              Object {
                "date_time": "2022-07-24 12:13:56 +0200",
              },
              Object {
                "date_time": "2022-07-25 20:49:10 +0200",
              },
            ],
            "mentor": Object {
              "name": "Max Mustermann",
              "time_zone": "-03:00",
            },
          }
        }
      />
    </div>
  </div>
</Fragment>
`;
